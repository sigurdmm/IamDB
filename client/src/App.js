import React from 'react';
import Header from './components/Header';
import 'App.less';

import Routes from './Routes';

const coverimages = [
  'https://media.giphy.com/media/XUTz8zpPF458I/giphy.gif',
  'https://media.giphy.com/media/hCm2X1kXxjyZq/giphy.gif',
  'https://media.giphy.com/media/xpBiaXFbVUtTa/giphy.gif',
  'https://media.giphy.com/media/PrGNf7O36heCs/giphy.gif',
  'https://media.giphy.com/media/26BRuo6sLetdllPAQ/giphy.gif',
  'https://media.giphy.com/media/3ohzdUi5U8LBb4GD4s/giphy.gif',
  'https://media.giphy.com/media/3oEjI1erPMTMBFmNHi/giphy.gif',
];

const selectRandomItem = list => list[Math.floor(Math.random() * list.length)];

const App = () => <>
  <div className="systemcover"
       style={{ backgroundImage: `url('${selectRandomItem(coverimages)}')` }}/>
  <div className="application application--light">
    <Header title="Hello world" className="application__header">
      <nav className="navigation">
        <a href="#" className="navigation__link">Test</a>
      </nav>
    </Header>
    <main className="application__content application__content--centered">
      <Routes/>
    </main>
    <footer className="application__footer">
      Copyright 2018
    </footer>
  </div>
</>;

export default App;
