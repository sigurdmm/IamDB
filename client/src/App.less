.application{
  background-color: #fff;
  box-sizing: border-box;

  margin: 0 auto;
  max-width: 60em;
  min-height: 100vh;

  padding: 1em;
  position: relative;


  width: 100%;
}

h1,
h2,
h3 {
  margin: 0;
}
