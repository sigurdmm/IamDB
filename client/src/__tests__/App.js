import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import App from '../App';

// Mock the store. Should only be necessary for this component.
// See this guide for more info. https://hackernoon.com/unit-testing-redux-connected-components-692fa3c4441c
const initialState = {};
const mockStore = configureMockStore();

describe('<App/>', () => {
  let store;
  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('should load properly', () => {
    const wrapper = shallow(<App store={store}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
