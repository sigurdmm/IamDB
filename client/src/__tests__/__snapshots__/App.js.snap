// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should load properly 1`] = `
<div>
  <Header
    title="Hello world"
  >
    <nav
      className="navigation"
    >
      <a
        className="navigation__link"
        href="#"
      >
        Test
      </a>
    </nav>
  </Header>
  <main>
    <Routes />
  </main>
  <footer>
    Copyright 2018
  </footer>
</div>
`;
